augur/cli/db.py

```diff
@@ -331,3 +332,188 @@ def check_pgpass_credentials(config):
+
+    augur_app.database.execute(insert_repo_sql)
+    return
+
+def get_value_from_fetch_data(data):
+    return data[0]
+
+def get_existing_repos(augur_app, repo_group_id):
+    repos_sql = s.sql.text("""
+            SELECT repo_name FROM repo
+            WHERE repo_group_id = '{}'
+        """.format(repo_group_id))
+
+    repos = augur_app.database.execute(repos_sql).fetchall()
+
+    if len(repos) == 0:
+        logger.info("Did not find any repositories stored in augur_database for repo_group_id {}\n".format(repo_group_id))
+
+    repos_list = list(map(get_value_from_fetch_data, repos))
+
+    return repos_list
+
+def get_org_id(augur_app, org):
+    org_id_sql = s.sql.text("""
+            SELECT repo_group_id FROM repo_groups
+            WHERE rg_name = '{}'
+        """.format(org))
+
+    org_id = augur_app.database.execute(org_id_sql).fetchone()[0]
+    if type(org_id) != int:
+        logger.info("Failed to get organization id from org {}".format(org))
+    return org_id
+
+def get_all_repos(org):
+    totalCount = get_repos_total_count(org)
+    maxCount = 100
+    nodes = []
+    after = ""
+
+    while totalCount > maxCount or totalCount <= maxCount and totalCount > 0:
+        totalCount = totalCount - maxCount
+        query = """
+        {
+            organization(login: "%s") {
+                repositories(first: %d%s) {
+                    edges {
+                        node {
+                            id
+                            name
+                            nameWithOwner
+                            url
+                            createdAt
+                        }
+                        cursor
+                    }
+                }
+            }
+        }
+        """ % (org, maxCount, after)
+
+        data = get_data(query, org)
+        if 'data' not in data:
+            logger.info("Could not get a list of all repositories. 'data' is missing in: {}\n".format(data))
+        if 'organization' not in data['data']:
+            logger.info("Could not get a list of all repositories. 'organization' is missing in: {}\n".format(data))
+        if 'repositories' not in data['data']['organization']:
+            logger.info("Could not get a list of all repositories. 'repositories' is missing in: {}\n".format(data))
+        if 'edges' not in data['data']['organization']['repositories']:
+            logger.info("Could not get a list of all repositories. 'edges' is missing in: {}\n".format(data))
+
+        nodes = nodes + data['data']['organization']['repositories']['edges']
+
+        l = len(data['data']['organization']['repositories']['edges'])
+        last_el = data['data']['organization']['repositories']['edges'][l-1]
+        after = ", after: \"" + last_el['cursor'] + "\""
+
+    data['data']['organization']['repositories']['edges'] = nodes
+
+    return data['data']['organization']['repositories']['edges']
+
+def get_repos_total_count(org):
```

Comment:
This function tells you if you have all the repos or not.  If you want to gather all of the repositories, FIRST, you have to get a list of repos that exist, compare that to the repos you have, and gather the delta.

Reply:
If you added a function to see if the repo count matches, you could skip the org.